src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier randomColor = new ColorSupplier();
+    private final int randomFigureType = random.nextInt(5);
+    private final int randomSide = random.nextInt() * 16;

```

Comment:
In this case, these values will be generated only once during object creation. So, you will have always same figure types and sides. `randomFigureType` and `randomSide` should be local scoped(if you really need those variables), but their magic numbers(random upper bounds etc.) should be constants.